models/interfaces.ts

```diff
@@ -7,15 +7,11 @@ export interface Detector {
   type: string;
   detector_type: string;
   name: string;
-  alert_conditions: AlertCondition[];
   enabled: boolean;
+  createdBy: string;
```

Comment:
Not blocking for a draft PR, but `createdBy` may need to support a `null` value when the security plugin isn't installed, or when the detector is being initially created. My understanding is that the backend populates the `createdBy` attribute.

Reply:
yeah for now I have kept everything as required since we don't have complete documentation about what is required vs optional. Will update once we have more clarity